### ISSUE TITLE
New Search Page feature overrides my own custom search static page

### ISSUE THREAD

**Issue by mtooke** [facebook/docusaurus#2945]
## 🐛 Bug Report

TL;DR: I believe the new search results page is overriding my own static search results page /search/.

I swizzled the searchBar component in alpha 40, in order to send search results to a static search results page that I created in React. I needed facets and filters, so I used InstantSearch. I created a page in React and used the InstantSearch React components. 

When I upgraded to the latest alpha (56), I notice that my search page no longer appears, though I do get a basic search bar there now and my results do show up when I use the field.

Unless I'm wrong, it looks like I can still swizzle and customize the search field, however I can no longer use my own search page? Unless there is a way to add facets and customize the search results, overriding the /search/ path really bums me out man. I don't see anything in the docs about this.

In short, I'm looking for a way to override this built-in search page.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes.

## To Reproduce

1. Create your own static search page that exports Search.
2. Implement Instantsearch.
3. Swizzle the searchbar to throw your results to the search page.

## Expected behavior

My own search page should appear, however I seem to get the new built-in search results instead?

![Screen Shot 2020-06-16 at 11 28 37 AM](https://user-images.githubusercontent.com/1414082/84794772-a8e12680-afc4-11ea-8bb7-a7cb0f331346.png)

## Actual Behavior
 The built-in search 

The empty search results page appears instead.

![Screen Shot 2020-06-16 at 11 29 20 AM](https://user-images.githubusercontent.com/1414082/84794784-ad0d4400-afc4-11ea-80d6-af4b42b24319.png)

## Your Environment

- Docusaurus version used: 40 is my current version, 55 had the new search page, though I was unable to get it working with 50 either?
- Version 79.0.3945.79, node:12.14.0
- Mac 10.14.6

## Reproducible Demo

https://github.com/mtooke/mtooke.github.io

This is my repo (site debranded for privacy). The search page is located in the src folder. This is currently working on Alpha 40.

**Comment by slorber** [comment]
Hi,

There are multiple possible solutions:


## Don't use `@docusaurus/theme-search-algolia`

The plugin is responsible to create a page `/search` page (currently not customizable)

Your `src/pages/search.js` wouldn't override it

Even if you don't use `@docusaurus/theme-search-algolia` you can still swizzle SearchBar like you do

You can disable this plugin by using something like `algolia: null` in classic preset. You can still pass algolia config using the `customFields` part of your site config.



## Use a different path

All this happens because you use `/search`, which conflicts with the algolia theme.

What about using another path, simply renaming `search.js` to something else?

Alternatively, we could allow to customize the path on the algolia theme so that it uses something else than `/search`


## We could allow swizzling the algolia search page

So you would sizzle the page instead of creating `src/pages/search.js`

current lt's not in the `theme` folder 

---

In your case, I'd recommend simply not using the algolia theme by disabling it, because it seems you won't use any line of code of this theme, so you'd better simply remove it.

**Comment by mtooke** [comment]
@slorber Thanks very much!!!

I was passing in my Algolia settings in my react page anyway, so setting algolia: null was all I needed to do. This solved my problem.

**Comment by slorber** [comment]
great :)
